src/aleph/vm/haproxy.py

```diff
@@ -0,0 +1,335 @@
+
+    # Track which servers we've processed to identify removals later
+    processed_servers = set()
+
+    # Process each mapping
+    for domain, target in mappings:
+        server_name = domain
+        processed_servers.add(server_name)
+
+        # Check if server already exists in mapping
+        if server_name in current_servers:
+            # FIXME : In the future, don't update the address if it hasn't changed'
+            # Update existing server
+            addr, port = target.split(":")
+            command = f"set server {backend_name}/{server_name} addr {addr} port {port}"
+            logger.info(f"Updating server: {command}")
+            response = send_socket_command(socket_path, command)
+            if response and "not found" in response:
+                logger.warning(f"Server not found: {server_name}, trying to add it")
+                # If server doesn't exist, add it
+                command = f"add server {backend_name}/{server_name} {target} weight {weight} maxconn 30"
+                logger.info(f"Adding server: {command}")
+                response = send_socket_command(socket_path, command)
+        else:
+            # Add new server
+            command = f"add server {backend_name}/{server_name} {target} weight {weight} maxconn 30"
+            logger.info(f"Adding server: {command}")
+            response = send_socket_command(socket_path, command)
+
+        # Check response
+        if response and "not found" in response:
+            logger.error(f"Error processing server {server_name}: {response}")
+        else:
+            command = f"enable server {backend_name}/{server_name}"
+            logger.info(f"Enable server: {command}")
+            response = send_socket_command(socket_path, command)
+            if response.strip() != "":
+                logger.info("Error enabling server Response")
+
+    # Remove servers that are not in the map file
+    servers_to_remove = set(current_servers) - processed_servers
+    if servers_to_remove:
+        logger.info(f"Removing {len(servers_to_remove)} servers no longer in map file")
+        for server_name in servers_to_remove:
+            logger.info(f"Removing server: {server_name}")
+            command = f"set  server {backend_name}/{server_name} state maint"
+            response = send_socket_command(socket_path, command)
+            command = f"del server {backend_name}/{server_name}"
+            response = send_socket_command(socket_path, command)
+            logger.info(f"Response: {response}")
+
+    return True
+
+
+async def fetch_list() -> list[dict]:
+    async with aiohttp.ClientSession() as client:
+        resp = await client.get(url=str(settings.DOMAIN_SERVICE_URL))
+        resp.raise_for_status()
+        instances = await resp.json()
+        if len(instances) == 0:
+            return []
+        return instances
+
+
+async def fetch_list_and_update2(socket_path, local_vms: list[str], force_update):
```

Comment:
Why this method is called `fetch_list_and_update2`, I think should be called `fetch_list_and_update`